framework/sandstone_tests.cpp

```diff
@@ -225,7 +227,26 @@ SandstoneTestSet::TestSet SandstoneTestSet::add_test_list(const char *fname, std
     std::vector<struct test_cfg_info> entries = load_test_list(list_file, this, cfg.ignore_unknown_tests, errors);
     if (!errors.empty()) return res;
     for (auto e : entries) {
-        test_set.push_back(e.test);
+        if (!e.attribute.empty()
+                && !std::ranges::equal(e.attribute, std::string("default"), [] (const char& a, const char& b) { return (std::tolower(a) == std::tolower(b)); })) {
+            /* the only attribute supported by the test list is time in
+             * milliseconds. */
+            uint64_t test_duration_ms;
+            std::string attr = e.attribute;
+            auto res = std::from_chars(attr.data(), attr.data() + attr.size(), test_duration_ms);
```

Comment:
Use `string_to_millisecs` from <sanstone_chrono.h>. That will also allow us to specify units instead of pure milliseconds.